gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -223,7 +223,7 @@ public static ClientContext create(StubSettings settings) throws IOException {
       transportChannelProvider = transportChannelProvider.withEndpoint(endpoint);
     }
     transportChannelProvider = transportChannelProvider.withUseS2A(endpointContext.useS2A());
-    if (transportChannelProvider.needsMtlsEndpoint()) {
+    if (transportChannelProvider.needsMtlsEndpoint() && endpointContext.mtlsEndpoint() != null) {
```

Comment:
Can we add a unit test for it? It would be great if we can mimic the Bigtable use case. Or maybe it is too complex to do so?

Reply:
I think it may be possible and I can take a look. I was just planning to make this a temporary fix (revert in the next cycle or two).